### ISSUE TITLE
Maxbotix sensors reporting erroneous distances

### ISSUE THREAD

**Issue by mdbartos** [open-storm/perfect-cell#53]
Maxbotix sensors are reporting erroneous distances after the latest update. See ARB003, ARB021, ARB022 and ARB023. This is possibly affecting the newest DFW nodes as well. In general, the reported distances are shorter than they should be. We haven't noticed any erroneous readings while testing in the lab. This suggests that the sensors might be having trouble "locking on".

My working theory is this: the sensors are not getting enough time to "ramp up" to the correct value.

During the last round of updates we made a couple changes:

- The CyDelay between power on and power off was reduced from 1500 to 800 ms.
- We fixed a bug that was causing the beginning of the uart buffer to be overwritten by the tail of the readings string.

In the new implementation, the parser extracts the first available reading, while the previous implementation (accidentally) grabbed the last reading, which gave the readings time to stabilize.

This could be fixed quickly by grabbing the last available reading. Probably the best way to fix the problem would be to continuously parse the readings until the moving average stabilizes (although this could be difficult to test in the lab).

**Comment by mdbartos** [comment]
On closer inspection, the error doesn't seem to be coming from the parser. The output string from the maxbotix is just sending strange values sometimes.

I ran debug with the maxbotix pointed towards the ceiling and it reported an output string like: 'R764 R765 R764 ...' which was incorrect. Then I tried stopping and ran the debugger again without changing anything and it reported 'R1604 R1604 R1605 ...' which was correct.

I wonder if this could be an issue with floating pins. Everything is set to strong drive right now.
